internal/kafka/internal/services/kafka.go

```diff
@@ -881,13 +905,33 @@ func (k *kafkaService) GetManagedKafkaByClusterID(clusterID string) ([]managedka
 		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "unable to list kafka requests")
 	}
 
+	enableKafkaExternalCertificate := k.kafkaTLSCertificateManagementService.IsKafkaExternalCertificateEnabled()
+
 	var res []managedkafka.ManagedKafka
 	// convert kafka requests to managed kafka
 	for _, kafkaRequest := range kafkaRequestList {
-		mk, err := buildManagedKafkaCR(kafkaRequest, k.kafkaConfig, k.keycloakService)
+		var certificate kafkatlscertmgmt.Certificate
+
+		if enableKafkaExternalCertificate { // only fetch certs when Kafka external certificates is enabled
+			certRequest := kafkatlscertmgmt.GetCertificateRequest{
+				TLSCertRef: kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef,
+				TLSKeyRef:  kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef,
+			}
+
+			var err error
+			certificate, err = k.kafkaTLSCertificateManagementService.GetCertificate(context.Background(), certRequest)
+			// TODO - gracefully handle errors so that we do not block reconciliation of others Kafkas
+			if err != nil {
+				return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to find kafka %q certificates", kafkaRequest.ID)
```

Comment:
should we log the error and continue here instead of returning? Currently, other Kafka requests don't get reconciled if an error happens.

Reply:
I'll prefer if we've a way to signal to the FSO that this MK is in erroneous state as part of the MK CR Spec. And the FSO will decide - basically skip the reconciliation on their end if they want to etc.

At the moment, if we do not include it in the list and the Kafka is already in the data plane - it'll cause a bunch of warnings in the Fleetshard side. But there could be other side effects that I do not know about. 

Also worth noting, if the error happens, it's likely going to be temporary (due to the backing storage service e.g AWS outage) and chances are that not one Kafka is going to be affect but all Kafkas.